README.md

```diff
@@ -1 +1,168 @@
+
+#### TypeScript Generics
+If a Class or Interfaces uses only one generic and the usages of the generic is obvious, then naming that generic `T` is sufficient. If a Class or Interface has more generics or when the usage is not obvious, you should use a more descriptive name. The same naming rules as for classes will apply then. 
+
+### Casing
+
+#### Classes, Interfaces, Types and Generics
+**PascalCase** Every individual word start with an upper case character, no underscores, no dashes.
+
+#### Functions, properties, arguments and variables
+**camelCase** Starts with a lower case character, every following individual word start with a upper case character, no underscores, no dashes. 
+
+#### Globally used constants
+**UPPER_CASE** Only use upper case characters, individual words must be separated with an underscore.
+
+#### CSS Class names
+**kabab-case** Only use lower case characters, individual words must be separated with a dash. 
+
+### File names
+If a file contains only one class, type or object, or when there is one main class, type or object with some helper classes, types or objects, the file should have the same name, in the same casing, as that (main) class, type or object. If a file contains a class, but only an instance of that class is exported, the file should have the same name as the class, but written in camelCase.
+
+If a file holds multiple classes, types and/or objects, and they are all more or less equal in importance, the file should have a name that describes all the classes, types and/or objects, written in camelCase.
+
+#### File extensions
+
+#### .js
+JavaScript only
+
+#### .ts
+TypeScript only
+
+####. jsx
+JavaScript with React JSX syntax. If a file has the .jsx extension, it must contain React code.
+
+####. tsx
+TypeScript with React JSX syntax. If a file has the .tsx extension, it must contain React code.
+
+## Coding
+Every function or class should do **one thing** (and do it good). If it needs to do more than one thing, split it up. Keep your files, classes and functions small. It’s okay to have a file with just a single line.
+
+#### Pure functions
+Prefer writing pure functions, which means they do not manipulate the input arguments or reference/manipulate global state. This makes your code better scalable and testable.
+
+#### Separate Logic From Configuration
+Write code that is reusable, scalable and testable.
+
+#### Do not repeat yourself (DRY)
+* Do not copy code to another place.
+* Avoid using the same string twice in a project.
+* Move shared logic to a shared place.
+* Make sure you do not have to adapt changes on multiple places.
+
+#### Do not use Magic Numbers
+See https://en.wikipedia.org/wiki/Magic_number_(programming)
+
+## Formatting
+All code within a project should have the same formatting. To enforce that we use [Prettier](https://prettier.io/).
+
+## Comments
```

Comment:
what about something like;
> if you're new to the project, or piece of code; when going over it, and trying to understand it, add explanatory comments on things that took you some time to figure out.

When you write code, it's hard to judge what is clear, and what is not.
But if you're new to the code, you're better equipped to understand what could use some clarification.

In addition, something like;
> When reviewing PRs, and someone asks why something is done a certain way, see if you can answer it with a code comment instead of a reply in the review tool (when applicable).